packages/core/src/lib/social.ts

```diff
@@ -0,0 +1,29 @@
+import { getSocialConnectorInstanceById } from '@/connectors';
+import { SocialUserInfo } from '@/connectors/types';
+import RequestError from '@/errors/RequestError';
+
+const getConnector = async (connectorId: string) => {
+  try {
+    return await getSocialConnectorInstanceById(connectorId);
```

Comment:
nit: could use Promise.catch directly? 

```
return await getSocialConnectorInstanceById().catch(e => {
 ...
})
```

Reply:
Lint rule prefer await and catch, so I won't change it for now.